src/container/init_builder.rs

```diff
@@ -45,12 +42,11 @@ impl InitContainerBuilder {
         let container_dir = self.create_container_dir()?;
         let spec = self.load_and_safeguard_spec(&container_dir)?;
 
-        unistd::chdir(&*container_dir)?;
```

Comment:
The removal of this line causes EBADF when trying to send the file descriptor of the pseudo terminal. The following setup_console_socket call fails to find the socket because the working directory is not the container directory.